addons/medical_feedback/functions/fnc_playInjuredSound.sqf

```diff
@@ -26,60 +26,35 @@ params [["_unit", objNull, [objNull]], ["_type", "hit", [""]], ["_severity", 0,
 if (!local _unit) exitWith {
     ERROR("Unit not local or null");
 };
-
 if !(_unit call EFUNC(common,isAwake)) exitWith {};
 
-private _timeOut = [TIME_OUT_HIT, TIME_OUT_MOAN] select (_type == "moan");
-
 // Handle timeout
+private _timeOut = [TIME_OUT_HIT, TIME_OUT_MOAN] select (_type == "moan");
 if (_unit getVariable [QGVAR(soundTimeout) + _type, -1] > CBA_missionTime) exitWith {};
 _unit setVariable [QGVAR(soundTimeout) + _type, CBA_missionTime + _timeOut];
 
-// Get sounds
-private _soundsNamespace = NAMESPACE_NULL;
-
-switch (toLower _type) do {
-    case ("hit"): {
-        _soundsNamespace = GVAR(HitScreamNamespace);
-    };
-    case ("moan"): {
-        _soundsNamespace = GVAR(InjuredMoanNamespace);
-    };
-};
-
 // Get units speaker
 private _speaker = speaker _unit;
-
 if (_speaker == "ACE_NoVoice") then {
     _speaker = _unit getVariable "ace_originalSpeaker";
 };
 
-private _sounds = _soundsNamespace getVariable _speaker;
-
-if (isNil "_sounds") then {
-    _sounds = _soundsNamespace getVariable (_soundsNamespace getVariable "#default");
-};
-
-if (isNil "_sounds") exitWith {
-    ERROR("No sounds for speaker and no default found");
-};
-
-// Get correct sound of the speaker
-_sounds = _sounds param [_severity, []];
-(selectRandom _sounds) params ["_sound", ["_volume", 1], ["_frequency", 1], ["_distance", 80]];
-
-if (isNil "_sound") exitWith {
-    ERROR("No sound for this speaker");
-};
-
-// Delete leading slash.
-if (_sound select [0, 1] == "\") then {
-    _sound = _sound select [1];
+// Fallback if speaker has no associated scream/moan sound
+if (isNull (configFile >> "CfgSounds" >> format ["ACE_moan_%1_low_1", _speaker])) then {
+    _speaker = "Male08ENG";
 };
 
-// Default file extension.
-if (_sound find "." == -1) then {
-    _sound = _sound + ".wss";
+// Select actual sound
+private _variation = ["low", "mid", "high"] select _severity;
+private _distance = if (_type == "hit") then {
+    [50, 60, 70] select _severity;
+} else {
+    [10, 15, 20] select _severity;
 };
+private _filter = toLower format ["(toLower configName _x) find 'ACE_%1_%2_%3' == 0", _type, _speaker, _variation];
+private _sounds = _filter configClasses (configFile >> "CfgSounds");
```

Comment:
`"(tolower configname _x) find 'ace_hit_male08eng_mid' == 0" configClasses (configFile >> "CfgSounds");`
is 3.9ms, probably worse with full modset (more CfgSounds entries)
I think we need to do caching at start

Reply:
switched to a while loop which takes about 0.05ms
could fully cache at game start, but this should be ok for now